arlo_server/ballots.py

```diff
@@ -114,20 +114,25 @@ def get_retrieval_list(election: Election, jurisdiction: Jurisdiction, round_id:
 def deserialize_interpretation(
     ballot_id: str, interpretation: JSONDict
 ) -> BallotInterpretation:
+    choices = ContestChoice.query.filter(
+        ContestChoice.id.in_(interpretation["choiceIds"])
+    ).all()
+    contest = Contest.query.get(interpretation["contestId"])
```

Comment:
I'm not crazy about repeating these queries here and in the validation functions. Though I don't think it's a big deal as long as SQLAlchemy/Posgresql does some caching. I thought about refactoring but wasn't sure it was worthwhile unless we know there's a real efficiency issue here.

Reply:
@benadida thoughts?

Reply:
I don't think you can count on the caching, but I also don't think this is worth optimizing – there is always at least a mild tradeoff between security & modularity. I'm ok with it.